source/extensions/filters/http/grpc_http1_bridge/http1_bridge_filter.cc

```diff
@@ -44,10 +45,20 @@ Http::FilterHeadersStatus Http1BridgeFilter::encodeHeaders(Http::ResponseHeaderM
     chargeStat(headers);
   }
 
-  if (!do_bridging_ || end_stream) {
+  if (!do_bridging_) {
+    return Http::FilterHeadersStatus::Continue;
+  }
+
+  response_headers_ = &headers;
+  ASSERT(response_headers_);
```

Comment:
No need to do the assert here. I meant it as a precondition for `Http1BridgeFilter::updateGrpcStatusAndMessage()` to catch errors in unit tests like
```c++
Http1BridgeFilter filter(config, context);
filter.updateGrpcStatusAndMessage(); // must blow up as the precondition for the function isn't satisfied.
```